### DISCUSSION TITLE
how do i make multiple fs requests in libuv?

### DISCUSSION THREAD

**Discussion by hxorasani** [libuv/libuv#3955] [Q&A]
I've been reading the docs and tried looking through the source code of projects using libuv and it's only made me more confused as to how to even setup two basic requests :((
```
uv_fs_open(uv_loop, &open_req, path1, O_RDONLY, 0, on_open);
uv_fs_open(uv_loop, &open_req, path2, O_RDONLY, 0, on_open);
```
where and how exactly am i supposed to define and cleanup `open_req` (and others... `read_req` etc) and am i using the `on_open` function correctly here?

I've tried the examples from the docs and they didn't really help me understand. An example of how to open two or more text files and print them using libuv would be a lot of help.

**Comment by trevnorris** [comment]
You need an `uv_fs_t` for each fs call. Then each needs to be cleaned up after the call is complete.

**Reply by cjihrig** [reply]
[Here](https://github.com/nodejs/node/blob/cad0ae702baedb24344be61be9e87c618a080471/src/util.cc#L199-L268) are some examples of synchronously reading and writing files. If you want to do it asynchronously, check out [this libuv test file](https://github.com/libuv/libuv/blob/2f33980a9141bd49cb2c34361854355b23b1e6fc/test/test-fs.c) or the [libuv book](https://nikhilm.github.io/uvbook/filesystem.html#reading-writing-files) (note - some parts of the book might be outdated, but that section should be ok).
